docs/guides/migrating-cra.md

```diff
@@ -0,0 +1,774 @@
+
+```js filename=entry.client.jsx lines=[2,4]
+import { RemixBrowser } from "@remix-run/react";
+import { hydrateRoot } from "react-dom/client";
+
+hydrateRoot(document, <RemixBrowser />);
+```
+
+## Creating your routes
+
+In CRA, the tool itself doesn't know or care about your routes. That's what React Router is there for, of course! By contrast, Remix is built on top of React Router, and routing is at the heart of how Remix works and where it shines. As such, we have some conventions on routing you'll want to follow.
+
+How you end up migrating your routes depends on how you've set them up in your existing app using React Router 6. Remix has two methods for defining routes:
+
+- Exporting route components from files inside the `routes` directory
+- Defining routes in your `remix.config.js` file
+
+Routes can also be defined in both places, and Remix will merge them to create all of your app's routes. [See our routing docs for more info.](../api/conventions)
+
+---
+
+Let's assume your app defines routes in `index.js` at the top of the rendering tree:
+
+```js filename=index.js
+import * as React from "react";
+import { createRoot } from "react-dom/client";
+import {
+  BrowserRouter,
+  Routes,
+  Route,
+} from "react-router-dom";
+
+import App from "./app";
+import Dashboard from "./dashboard";
+import Expenses from "./expenses";
+import Invoices from "./invoices";
+import UserLayout from "./user-layout";
+import UserProfile from "./user-profile";
+import UserSales from "./user-sales";
+
+const root = createRoot(document.getElementById("root"));
+root.render(
+  <BrowserRouter>
+    <Routes>
+      <Route path="/" element={<App />}>
+        <Route index element={<Dashboard />} />
+        <Route path="expenses" element={<Expenses />} />
+        <Route path="invoices" element={<Invoices />} />
+        <Route path=":userId" element={<UserLayout />}>
+          <Route path="profile" element={<UserProfile />} />
+          <Route path="sales" element={<UserSales />} />
+        </Route>
+      </Route>
+    </Routes>
+  </BrowserRouter>
+);
+```
+
+In Remix, you never need to explicitly render the `<BrowserRouter />`, `<Routes />` or `<Route />`. This route hierachy can be replicated by:
+
+- rendering the `App` component in special route file called `root.jsx` in your `appDirectory`
+- moving the other route components into the `routes` directory
+- updating file names to follow our [file name conventions](../api/conventions#file-name-conventions)
+
+So you'll end up with a file structure that looks like this:
+
+```
+app/
+├── routes/
+│   ├── $userId/
+│   │   ├── profile.jsx
+│   │   ├── sales.jsx
+│   └── expenses.jsx
+│   └── index.jsx
+│   └── invoices.jsx
+│   └── $userId.jsx
+└── root.jsx
+```
+
+### The `root` route
+
+The root route (or the "root root" if you're Wes Bos) is responsible for providing the structure of the application. Its default export is a component that renders the full HTML tree.
```

Comment:
@wesbos, just want to double-check with you that you're ok with this joke 😅

Reply:
haha yep, I didn't realize I was known for the way I say Route :)